### PR TITLE
project: optimize loading speed and SEO

### LINKED ISSUE #111: Optimize pages loading speed and resources loading.

### LINKED ISSUE #112: Optimize pages for referencing

### PR DESCRIPTION
**Pull request by sebastiendeleze**
* Optimizes loading speed and SEO after running a Lighthouse test.
* Removes webpack entry for `app` and moves `app.js` file in static files.
* Removes useless dependencies `popper.js` and `jquery`.
* Triggers dropdowns in vanilla javascript instead of using jQuery.
* Removes unsused local Roboto font.
* Loads Roboto font and Font-awesome externally.
* Closes #111.
* Closes #112.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

**Review by BadrAly — APPROVED** [review]

File: sonar/theme/templates/sonar/page.html [hunk 1]
```diff
@@ -1,148 +1,207 @@
...
-    {%- block outer_body %}
-      {%- block browserupgrade %}
-        <!--[if lt IE 8]>
+<html {% if html_css_classes %} class="{{ html_css_classes|join(' ') }}" {% endif %}
+  lang="{{ current_i18n.locale.language|safe }}" dir="{{ current_i18n.locale.text_direction }}">
+
+<head>
+  {% if config.ENV == 'production' %}
+  <!-- Global site tag (gtag.js) - Google Analytics -->
+  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-1831395-15"></script>
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:30]: Make google id on Invenio config (UA-1831395-15)

File: sonar/theme/templates/sonar/page.html [hunk 2]
```diff
@@ -1,148 +1,207 @@
...
+  <script>
+    window.dataLayer = window.dataLayer || [];
+    function gtag() { dataLayer.push(arguments); }
+    gtag('js', new Date());
+
+    gtag('config', 'UA-1831395-15');
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:36]: Make google id on Invenio config (UA-1831395-15)

File: sonar/theme/static/js/app.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/*
+  Swiss Open Access Repository
+  Copyright (C) 2019 RERO
```
Comment (Garfield-fr) [inline sonar/theme/static/js/app.js:3]: `Copyright (C) 2021 RERO`

File: sonar/theme/templates/sonar/frontpage.html [hunk 4]
```diff
@@ -21,42 +21,46 @@
...
+    <div class="row justify-content-center">
+      <div class="col-8 col-xs-6 col-lg-3 py-3 py-sm-5 text-center">
+        <a
+          href="{{ url_for('documents.index', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}">
+          {% if g.get('organisation', {}).get('isDedicated') %}
+          {% set thumbnail = g.organisation | record_image_url %}
+          {% if thumbnail %}
+          <img src="{{ thumbnail }}" class="img-fluid" alt="{{ 'Organisation logo' }}">
+          {% else %}
+          <h1>{{g.organisation.name}}</h1>
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/frontpage.html:34]: missing space after `{` and before `}`

File: sonar/theme/templates/sonar/frontpage.html [hunk 5]
```diff
@@ -21,42 +21,46 @@
...
+    </div>
+    <div class="row justify-content-center">
+      <div class="col-lg-8 text-right my-4">
+        <form class="justify-content-end"
+          action="{{ url_for('documents.search', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}"
+          role="search">
+          <div class="row">
+            <div class="col-9 col-md-11">
+              <input class="form-control form-control-lg mr-2" type="search"
+                placeholder="{{_('Search publications, authors, projects...')}}" aria-label="Search" name="q">
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/frontpage.html:50]: missing space after `{` and before `}`

File: sonar/theme/templates/sonar/frontpage.html [hunk 6]
```diff
@@ -69,32 +73,27 @@ <h6 class="m-0 mt-1">
...
 {%- block body %}
-  <div class="row justify-content-between">
-    <div class="col-12 col-lg-5 pr-5">
-      <h3 class="mb-4">{{_('The project')}}</h2>
-      <p>{{_('The SONAR project aims to create a scholarly archive that collects, promotes and preserves the publications of authors affiliated with Swiss public research institutions.')}}</p>
-      <p>{{_('Further info on the project page')}} <a href="https://sonar.ch/project" target="_blank">sonar.ch</a>.</p>
+<div class="row justify-content-between">
+  <div class="col-lg-7">
+    <h5 class="home-text">
+      {{_('The Swiss Open Access Repository is an archive of scholarly publications. It collects and promotes open access publications by authors affiliated with Swiss public research institutions.')}}
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/frontpage.html:79]: missing space after `{` and before `}`

File: sonar/theme/templates/sonar/frontpage.html [hunk 7]
```diff
@@ -69,32 +73,27 @@ <h6 class="m-0 mt-1">
...
+    </h5>
+  </div>
+  <div class="col-lg-4 mt-3 mt-lg-0">
+    <h3 class="mb-4">{{_('Follow us')}}</h3>
+    <div class="row align-items-center mt-3">
+      <div class="col-2">
+        <i class="fa fa-github fa-2x"></i>
+      </div>
+      <div class="col-10">{{_('Source code on')}} <a href="https://github.com/rero/sonar" target="_blank" rel="noopener">Github</a>
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/frontpage.html:88]: missing space after `{` and before `}`

File: sonar/theme/templates/sonar/page.html [hunk 8]
```diff
@@ -1,148 +1,207 @@
...
+  </script>
+  {% endif %}
+  {%- block head %}
+  {%- block head_meta %}
+  <meta charset="utf-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1">
+  {%- if description %}
+  <meta name="description" content="{{ description }}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:45]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 9]
```diff
@@ -1,148 +1,207 @@
...
+  {% endif %}
+  {%- if keywords %}
+  <meta name="keywords" content="{{ keywords }}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:48]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 10]
```diff
@@ -1,148 +1,207 @@
...
+  {% endif %}
+  {%- if config.get('THEME_GOOGLE_SITE_VERIFICATION', None) %}
+  {%- for google_id in config.THEME_GOOGLE_SITE_VERIFICATION %}
+  <meta name="google-site-verification" content="{{google_id}}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:52]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 11]
```diff
@@ -1,148 +1,207 @@
...
+  <link rel="apple-touch-icon" sizes="152x152" href="/static/apple-icon-152x152.png">
+  <link rel="apple-touch-icon" sizes="180x180" href="/static/apple-icon-180x180.png">
+  <link rel="icon" type="image/png" sizes="192x192" href="/static/android-icon-192x192.png">
+  <link rel="icon" type="image/png" sizes="32x32" href="/static/favicon-32x32.png">
+  <link rel="icon" type="image/png" sizes="96x96" href="/static/favicon-96x96.png">
+  <link rel="icon" type="image/png" sizes="16x16" href="/static/favicon-16x16.png">
+  <link rel="manifest" href="/static/manifest.json">
+
+  {%- if keywords %}
+  <link rel="canonical" href="{{ canonical_url }}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:82]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 12]
```diff
@@ -1,148 +1,207 @@
...
+  {% endif %}
+
+  {%- block head_links_langs %}
+  {%- if alternate_urls %}
+  {%- for alt_ln, alternate_url in alternate_urls.items() %}
+  <link rel="alternate" hreflang="{{ alt_ln }}" href="{{ alternate_url }}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:88]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 13]
```diff
@@ -1,148 +1,207 @@
...
+  {%- endfor %}
+  {%- endif %}
+  {%- endblock %}
+
+  {%- block head_apple_icons %}
+  {%- for size in [144, 114, 72, 57] %}
+  {%- set icon_name = 'apple-touch-icon-%d-precomposed.png' | format(size) %}
+  <link rel="apple-touch-icon-precomposed" sizes="{{ size }}x{{ size }}"
+    href="{{ url_for('static', filename=icon_name) }}" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:97]: ` />` to `>`

File: sonar/theme/templates/sonar/page.html [hunk 14]
```diff
@@ -1,148 +1,207 @@
...
+
+  {%- block css %}
+  {{ webpack[get_specific_theme()] }}
+  <link rel="preconnect" href="https://fonts.gstatic.com">
+  <link
+    href="https://fonts.googleapis.com/css2?family=Roboto+Condensed:wght@300;700&family=Roboto:wght@300;700&display=swap"
+    rel="stylesheet">
+  <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css"
+    integrity="sha512-SfTiTlX6kk+qitfevl/7LibUOeJWlt9rbyDn92a1DqWOw9vWG2MFoays0sgObmWazO5BQPiFucnnEAjpAB+/Sw=="
+    crossorigin="anonymous" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page.html:111]: ` />` to `>`

File: sonar/theme/templates/sonar/page_cover.html [hunk 15]
```diff
@@ -21,66 +22,65 @@
...
-              </a>
-            {% endif %}
-          {%- endblock %}
+  <div class="container">
+    <div class="row justify-content-center">
+      <div class="col-12 col-lg-6 text-center">
+        {%- block brand %}
+        {%- if config.THEME_LOGO %}
+        <a href="/">
+          <img class="text-center logo" src="{{ url_for('static', filename=config.THEME_LOGO)}}"
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page_cover.html:39]: space before `}` at the end

File: sonar/theme/templates/sonar/page_cover.html [hunk 16]
```diff
@@ -21,66 +22,65 @@
...
+            alt="{{ _(config.THEME_SITENAME) }}" height="70" />
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/page_cover.html:40]: ` />` to `>`

File: sonar/theme/templates/sonar/partial/navbar.html [hunk 17]
```diff
@@ -1,84 +1,80 @@
...
-      <i class="fa fa-lock"></i>
+    {% if page != 'home' %}
+    <a class="navbar-brand" href="{{ url_for('documents.index', view=view_code) }}">
+      {% if g.get('organisation', {}).get('isDedicated') %}
+      {% set thumbnail = g.organisation | record_image_url %}
+      {% if thumbnail %}
+      <img src="{{ thumbnail }}" alt="{{ _('Organisation logo') }}" height="50"
+        class="d-inline-block align-top mr-3 my-2" alt="">
+      {% else %}
+      {{g.organisation.name}}
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/partial/navbar.html:28]: space after `{` and before `}`

File: sonar/theme/templates/sonar/partial/navbar.html [hunk 18]
```diff
@@ -1,84 +1,80 @@
...
-      <form action="{{ url_for('documents.search', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}"
+      {% if page != 'home' %}
+      <form
+        action="{{ url_for('documents.search', view=view_code if g.get('organisation', {}).get('isDedicated') else config.SONAR_APP_DEFAULT_ORGANISATION) }}"
         class="form-inline my-2 my-lg-0 ml-lg-3">
-        <input name="q" class="form-control {{ 'form-control-sm' if g.get('organisation') else '' }} mr-sm-2" type="search" placeholder="{{_('Search')}}"
-          aria-label="{{_('Search')}}" value="{{ request.args.get('q', '') }}">
-        <button class="btn btn-outline-light {{ 'btn-sm' if g.get('organisation') else '' }} my-2 my-sm-0" type="submit">
+        <input name="q" class="form-control {{ 'form-control-sm' if g.get('organisation') else '' }} mr-sm-2"
+          type="search" placeholder="{{_('Search')}}" aria-label="{{_('Search')}}"
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/partial/navbar.html:49]: space after `{` and before `}`

File: sonar/theme/templates/sonar/partial/navbar.html [hunk 19]
```diff
@@ -1,84 +1,80 @@
...
       <span class="navbar-text text-secondary">{{ _('Administration')|upper }}</span>
       {% endif %}
-      {% endif %}
       <ul class="navbar-nav ml-auto">
         {%- if not current_user.is_authenticated %}
-        <li class="nav-item px-3 d-none d-lg-block">
-          <a class="nav-link" href="{{url_for_security('login', next=request.path)}}" title="{{ _('Log in') }}">
-            <i class="fa fa-lock"></i>
+        <li class="nav-item px-3">
+          <a class="nav-link text-light" href="{{url_for_security('login', next=request.path)}}" title="{{ _('Log in') }}">
```
Comment (Garfield-fr) [inline sonar/theme/templates/sonar/partial/navbar.html:63]: space after `{` and `}`

**Review by Garfield-fr — APPROVED** [review]

File: sonar/theme/static/js/app.js (hunk 3 above)
Comment (sebastiendeleze) [reply sonar/theme/static/js/app.js:3]: I will do that in another PR.

**Pull request merged**